_posts/2025-04-30-#31981.md

```diff
@@ -7,20 +7,55 @@ authors: [Sjors]
 components: ["mining"]
 host: ismaelsadeeq
 status: upcoming
-commit:
+commit: e6e170cf6c67a56b9c14cece66fdc4fab5f3ec6b
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+### Background
+
+[StratumV2 Sv2](https://github.com/stratum-mining/sv2-spec/tree/main) is a standard protocol that improves upon the StratumV1 protocol. 
+Sv2 has several benefits, one of which is allowing individual miners connected to a mining pool to build their own block templates. This further decentralizes Bitcoin mining and improves the censorship resistance of Bitcoin transactions.
+
+The Sv2 standard defines various roles, one of which is the Template Provider (TP), explained in the [template distribution](https://github.com/stratum-mining/sv2-spec/blob/main/07-Template-Distribution-Protocol.md) section of the spec. This role pushes block template updates to miners whenever the chain tip is updated or when mempool fees have increased sufficiently.
+In the Sv2 spec, it is specified that Bitcoin Core should fulfill the TP role, which means it has to implement the Sv2 spec in order to communicate with other roles.
+
+There have been attempts to add the Sv2 TP role to Bitcoin Core in [#23049](https://github.com/bitcoin/bitcoin/pull/23049), [#27854](https://github.com/bitcoin/bitcoin/pull/27854), and [#29432](https://github.com/bitcoin/bitcoin/pull/29432), all of which failed to gather rough from Bitcoin Core contributors.
+Major objections were technical and conceptual, questioning whether the Sv2 Noise protocol encryption, messaging layer, and other specifications should be supported in Bitcoin Core, which would require a large code addition, review effort, and maintenance burden. Most of these concerns can be read in the deceased PR's comments.
+
+Alongside this, there is an ongoing project [multi process Bitcoin](https://github.com/bitcoin/bitcoin/issues/28722) which introduced a new Inter-Process Communication (IPC) scheme, which allows the bitcoind process to communicate with other processes. Noting this, Bitcoin Core contributors suggested creating a mining IPC interface such that an external process, a TP, would connect to it and receive push notifications of new block templates as specified in the spec. This template provider would then forward the block to the [Job Declarator](https://github.com/stratum-mining/sv2-spec/blob/main/06-Job-Declaration-Protocol.md).
+
+The mining interface has been implemented in Bitcoin Core with two classes `BlockTemplate` and `Mining`, which have several methods needed by the Sv2 TP, effectively allowing projects like [Sjors TP](https://github.com/Sjors/bitcoin/pull/49) to utilize it.
+
+This approach has had some latency issues compared to when Bitcoin Core directly acts as the TP.
+
+### Motivation
```

Comment:
The paragraph that follows is not really a motivation, so probably either the content or the header should be changed?